vcd/resource_vcd_vm_internal_disk.go

```diff
@@ -0,0 +1,477 @@
+package vcd
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"log"
+	"strings"
+	"text/tabwriter"
+)
+
+func resourceVmInternalDisk() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVmInternalDiskCreate,
+		Read:   resourceVmInternalDiskRead,
+		Update: resourceVmInternalDiskUpdate,
+		Delete: resourceVmInternalDiskDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceVcdVmInternalDiskImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"vapp_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The vApp this VM internal disk belongs to",
+			},
+			"vm_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "VM in vApp in which internal disk is created",
+			},
+			"allow_vm_reboot": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				Description: "	Powers off VM when changing any attribute of an IDE disk or unit/bus number of other disk types, after the change is complete VM is powered back on. Without this setting enabled, such changes on a powered-on VM would fail.",
+			},
+			"bus_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"ide", "parallel", "sas", "paravirtual", "sata"}, true),
```

Comment:
I see case is ignored for this validator - won't there be any plan or API issues?

Reply:
no -> `AdapterType:         internalDiskBusTypes[d.Get("bus_type").(string)],`

Reply:
I mean exactly. The name can be in any case `ide`, `IDE` and even `iDe`. But here is what I get if I enter IDE:

```
vcd_vm_internal_disk.disk2: Creating...

Error: error updating VM disks: [VM web1 Id urn:vcloud:vm:3f5ea03e-c205-43e4-89bc-71a282485ac0] disk settings missing required field: adapter type

  on egw.tf line 83, in resource "vcd_vm_internal_disk" "disk2":
  83: resource "vcd_vm_internal_disk" "disk2" {
```

Reply:
Lower case only accepted now.